seed/views/main.py

```diff
@@ -2340,17 +2342,30 @@ def parse_energy_template(request):
 
     import_file = ImportFile.objects.get(pk=file_id)
     file_path = str(import_file.file)
+    file_path = default_storage.open(file_path, 'r')
+
     _log.info(file_path)
 
-    json_data = energy_template_process.parse_energy_template(file_path)
+    json_data = energy_template_process.parse_energy_template_file(file_path)
 
     green_button_data_analyser.data_analyse(json_data, 'Energy Template')
 
+    file_path.close()
     res = {}
     res['status'] = 'success'
     return res
 
 
+def valid_timestamp(timestamp):
+    res = {}
+    match_obj = re.match(r'\d+', timestamp)
```

Comment:
I compared the performance between regex checking and convert to int 1 million times respectively, seems regex checking takes one more second to finish. I updated the code to try and typecast and catch the failure.